duckdb-vortex/src/expr/expr.cpp

```diff
@@ -0,0 +1,216 @@
+	scalar->mutable_value()->set_null_value(google::protobuf::NULL_VALUE);
+	return scalar;
+}
+
+vortex::scalar::Scalar *into_vortex_scalar(Arena &arena, Value &value, bool nullable) {
+	auto scalar = Arena::Create<vortex::scalar::Scalar>(&arena);
+	auto dtype = into_vortex_dtype(arena, value.type().id(), nullable);
+	scalar->set_allocated_dtype(dtype);
+
+	switch (value.type().id()) {
+	case LogicalTypeId::INVALID:
+	case LogicalTypeId::SQLNULL: {
+		scalar->mutable_value()->set_null_value(google::protobuf::NULL_VALUE);
+		return scalar;
+	}
+	case LogicalTypeId::BOOLEAN: {
+		auto boolean = new vortex::dtype::Bool();
+		boolean->set_nullable(nullable);
+		scalar->mutable_value()->set_bool_value(value.GetValue<bool>());
+		return scalar;
+	}
+	case LogicalTypeId::TINYINT:
+		scalar->mutable_value()->set_int8_value(value.GetValue<int8_t>());
+		return scalar;
+	case LogicalTypeId::SMALLINT:
+		scalar->mutable_value()->set_int16_value(value.GetValue<int16_t>());
+		return scalar;
+	case LogicalTypeId::INTEGER:
+		scalar->mutable_value()->set_int32_value(value.GetValue<int32_t>());
+		return scalar;
+	case LogicalTypeId::BIGINT:
+		scalar->mutable_value()->set_int64_value(value.GetValue<int64_t>());
+		return scalar;
+	case LogicalTypeId::UTINYINT:
+		scalar->mutable_value()->set_uint8_value(value.GetValue<uint8_t>());
+		return scalar;
+	case LogicalTypeId::USMALLINT:
+		scalar->mutable_value()->set_uint16_value(value.GetValue<uint16_t>());
+		return scalar;
+	case LogicalTypeId::UINTEGER:
+		scalar->mutable_value()->set_uint32_value(value.GetValue<uint32_t>());
+		return scalar;
+	case LogicalTypeId::UBIGINT:
+		scalar->mutable_value()->set_uint64_value(value.GetValue<uint64_t>());
+		return scalar;
+	default:
+		break;
+	}
+	throw Exception(ExceptionType::NOT_IMPLEMENTED, "into_vortex_scalar", {{"id", value.ToString()}});
+}
+
+void set_column(const string &s, vortex::expr::Expr *column) {
+	auto get_item = new vortex::expr::Kind_GetItem();
+	get_item->mutable_path()->assign(s);
+	auto kind = column->mutable_kind();
+	kind->set_allocated_get_item(get_item);
+	column->add_children()->mutable_kind()->set_allocated_identity(new vortex::expr::Kind_Identity());
+}
+
+vortex::expr::Expr *table_expression_into_expr(Arena &arena, TableFilter &filter, string &column_name) {
+	auto expr = Arena::Create<vortex::expr::Expr>(&arena);
+	switch (filter.filter_type) {
+	case TableFilterType::CONSTANT_COMPARISON: {
+		auto &constant_filter = filter.Cast<ConstantFilter>();
+		auto bin_op = into_binary_operation(constant_filter.comparison_type);
+		auto value = into_vortex_scalar(arena, constant_filter.constant, true);
+
+		auto column = expr->add_children();
+		set_column(column_name, column);
+
+		auto constant = expr->add_children()->mutable_kind();
+		auto literal = constant->mutable_literal();
+		literal->set_allocated_value(value);
+
+		expr->mutable_kind()->set_binary_op(bin_op);
+		expr->set_id("binary");
```

Comment:
Should these ids be defined somewhere higher up as a constant?

Reply:
Could be a map from enum variants to const strings.

Reply:
I think setting explicitly, make more sense since I expect the variants to disappear soon